java/src/main/scala/com/spotify/featran/java/SerializableFunction.scala

```diff
@@ -15,10 +15,8 @@
  * under the License.
  */
 
-package com.spotify.featran.java;
+package com.spotify.featran.java
 
-import java.io.Serializable;
+import java.util.function.{Function => JFunction}
 
-public interface SerializableFunction<InputT, OutputT> extends Serializable {
-  OutputT apply(InputT input);
-}
+trait SerializableFunction[I, O] extends JFunction[I, O] with Serializable
```

Comment:
binary compatibility relies on compilation to translate to java interface bytecode.